spec/integration/criteria/raw_value_spec.rb

```diff
@@ -0,0 +1,397 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+describe 'Queries with Mongoid::RawValue criteria' do
+  before { Time.zone = 'UTC'}
+  let(:now_utc) { Time.utc(2020, 1, 1, 16, 0, 0, 0) }
+  let(:today) { Date.new(2020, 1, 1) }
+
+  let(:labels) do
+    [ Label.new(age: 12), Label.new(age: 16) ]
+  end
+
+  let!(:band1) { Band.create!(name: '1', likes: 0, rating: 0.9, sales: BigDecimal('90'), decibels: 20..80, founded: today, updated_at: now_utc) }
+  let!(:band2) { Band.create!(name: '2', likes: 1, rating: 1.0, sales: BigDecimal('100'), decibels: 30..90, founded: today, updated_at: now_utc + 1.days) }
+  let!(:band3) { Band.create!(name: '3', likes: 1, rating: 2.2, sales: BigDecimal('220'), decibels: 40..100, founded: today + 1.days, updated_at: now_utc + 2.days) }
+  let!(:band4) { Band.create!(name: '3', likes: 2, rating: 3.1, sales: BigDecimal('310'), decibels: 50..120, founded: today + 1.days, updated_at: now_utc + 3.days) }
+  let!(:band5) { Band.create!(name: '4', likes: 3, rating: 3.1, sales: BigDecimal('310'), decibels: 60..150, founded: today + 2.days, updated_at: now_utc + 3.days, labels: labels) }
+
+  context 'Mongoid::RawValue<String> criteria' do
+
+    context 'Integer field' do
+      it 'does not match objects' do
+        expect(Band.where(likes: Mongoid::RawValue('1')).to_a).to eq []
```

Comment:
The tests should assert that RawValue works, i.e. that it returns query results. You can use `Band.collection.insert_many` to populate the database with values that bypass Mongoid field logic. Do you have positive assertions on this field in other tests? It is unclear which tests that would be if so.

Reply:
Done. I've used the `Band.collection.insert_one` approach you advocated.